### PR TITLE
Fix compatibility with `pymc3` and `pymc==4.0.0b2`

### LINKED ISSUE #3: Add compatibility with Aesara
Currently the imports and particularly https://github.com/JuBiotech/murefi/blob/master/murefi/symbolic.py are limited to `theano-pymc` as the backend for symbolic graph computation.

`calibr8` is already compatible with Aesara and it's probably not too hard to make `murefi` compatible as well.

### PR DESCRIPTION
**Pull request by michaelosthege**
* The test pipeline was refactored to test all currently supported combinations.
* For simplicity the PyMC/sunode integration tests were removed from the release pipeline.
* Dependencies were moved into `requirements.txt` and `requirements-dev.txt` like we have in other projects.
* Some type hints were added alongside a fix of an incorrectly typed dict element assignment.

Closes #3

### CODE REVIEW

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -17,30 +18,28 @@ jobs:
...
     - name: Lint with flake8
       run: |
         # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings
         flake8 . --count --exit-zero --statistics
-    - name: Test with pytest
+    - name: Test without PyMC
       run: |
         pytest --cov=./murefi --cov-append --cov-report xml --cov-report term-missing murefi/tests.py
```
Comment (lhelleckes) [inline .github/workflows/pipeline.yml:30]: Don't you repeat this three times now for the different pymc-versions in the matrix?

File: murefi/symbolic.py [hunk 2]
```diff
@@ -1,39 +1,51 @@
...
+        # Theano-PyMC 1.1.2
+        import theano as _backend
+        from theano.graph.op import Op
+        from theano.graph.basic import Apply, Variable
+        import theano.tensor as at
+    except ModuleNotFoundError:
+        _backend = calibr8.utils.ImportWarner("aesara")
+        at = calibr8.utils.ImportWarner("aesara")
+        Op = object
+        Apply: TypeAlias = Any
```
[the remark this answers is not in the document]
Comment (michaelosthege) [reply murefi/symbolic.py:29]: The `Op` must be defined for the `class IntegrationOp(Op)` definition to work.

`Variable` is used in some signatures and must be defined for the type hints. But here Aesara/Theano are not available, so I created them as aliases for the `Any` type.
This makes type checkers such as `mypy` happy.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -19,23 +19,11 @@ jobs:
...
       run: |
-        pip install -e . 
-        pip install flake8 pytest pytest-cov twine wheel
-    - name: Test with pytest
+        pip install -e .
+        pip install -r requirements-dev.txt
+    - name: Test without PyMC
       run: |
         pytest --cov=./murefi --cov-append --cov-report xml --cov-report term-missing murefi/tests.py
-    - name: Install PyMC3 and PyGMO
```
[the remark this answers is not in the document]
Comment (lhelleckes) [reply .github/workflows/release.yml:27]: @michaelosthege You didn't comment this one.

[the remark this answers is not in the document]
Comment (michaelosthege) [reply .github/workflows/release.yml:27]: Overlooked it, but this is in the PR description. The CI runs on main and we don't make a release when it is red, so there's little benefit from another pymc/sunode test

**Review by lhelleckes — APPROVED** [review]
Looking good except you dind't say why in teh release job no PyMC testing is applied.

**Pull request merged**